go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -3519,6 +3522,61 @@ func (s *VtctldServer) SetShardTabletControl(ctx context.Context, req *vtctldata
 	}, nil
 }
 
+// SetVtorcEmergencyReparent enable/disables the use of EmergencyReparentShard in VTOrc recoveries for a given keyspace or keyspace/shard.
+func (s *VtctldServer) SetVtorcEmergencyReparent(ctx context.Context, req *vtctldatapb.SetVtorcEmergencyReparentRequest) (resp *vtctldatapb.SetVtorcEmergencyReparentResponse, err error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.SetVtorcEmergencyReparent")
+	defer span.Finish()
+
+	defer panicHandler(&err)
+
+	span.Annotate("keyspace", req.Keyspace)
+	span.Annotate("shard", req.Shard)
+	span.Annotate("disable", req.Disable)
+
+	ctx, unlock, lockErr := s.ts.LockKeyspace(ctx, req.Keyspace, "SetVtorcEmergencyReparent")
+	if lockErr != nil {
+		err = lockErr
+		return nil, err
+	}
+
+	defer unlock(&err)
+
+	if req.Shard != "" && req.Shard != "-" {
```

Comment:
I have seen single shard keyspaces referred to as shard `0`. Should that be included here?

Reply:
@ejortegau technically `0` is a single, non-ranged shard. And `-` is actually referring to the entire shard range

So, no I believe this logic is correct as-is. Considering `0` to be a whole keyspace has consequences on splits